### DISCUSSION TITLE
Monthly timer

### DISCUSSION THREAD

**Discussion by Likemike24** [arendst/Tasmota#21369] [Q&A] [answered]
Hello,

I am looking for a solution to the following problem.

Tasmota should always switch on a socket on the first Saturday of a month at 10 am. This should then remain active for 1 hour and then switch off again.

Is it possible to implement this with rules and timers?

**Comment by sfromis** [comment]
You could create a timer for every saturday, which could then create an event with `%timestamp%` as parameter, giving you something like 2024-05-06T14:59:09. Then you could create 7 triggers matching on string contains `$|` to check if the value contains strings like `-01T` up to `-07T` to catch the first saturday. Yes, that's a bit of mess, but rules does have limited flexibility.

For switching off again after 1 hour, the command `TimedPower 3600000,on` will switch on "now", and off again after 1 hour (the time value is in milliseconds).

If using ESP32, you could have a single `cron` to catch the first saturday every month, at 10:00. That would not have to be on the plug itself, could be on another device with ESP32.

**Reply by Likemike24** [reply]
I´m sorry, but I´m really bad at programming. Could you give me a hint how to start with some code please? The device uses a ESP8266 if I´m correct.

**Reply by sfromis** [reply]
It is no problem if you are bad at programming, as rules are pretty different from a programming language. Still a learning curve, and many examples on https://tasmota.github.io/docs/Rules/

**Reply by Likemike24** [reply]
Ok, thank you for pointing out. I´ll try to solve the task on my own.

**Reply by Likemike24** [reply, marked as answer]
Solved it:
I´m using a NOUS A1T. Since this device seems to not support the TimedPower parameter I had to use PulseTime for turning off the plug after 1 Minute.

For 60 seconds timed power off enter in console (Offset 100 + 60 seconds = 60 seconds):
`PulseTime 160`

The set a repeating timer for every saturday via GUI and set Action RULE (in this case I´m using Timer1.
Then execute the following code after Timer1 expires to check current date and power on if date fits the first 7 days of the week:
`Rule1 on clock#Timer=1 do backlog event checkdate=%timestamp% endon on event#checkdate$|-01T do Power1 1 endon on event#checkdate$|-02T do Power1 1 endon on event#checkdate$|-03T do Power1 1 endon on event#checkdate$|-04T do Power1 1 endon on event#checkdate$|-05T do Power1 1 endon on event#checkdate$|-06T do Power1 1 endon on event#checkdate$|-07T do Power1 1 endon`

**Reply by sfromis** [reply]
Great, that was what I meant. :smile: 

When it comes to `TimedPower`, that is a pretty new command, meaning that it is only available with latest Tasmota version, also on a Nous device.

**Reply by Likemike24** [reply]
Thank you for the information and pointing me in the right direction!
